public/mobile-app/src/lib/agenda.ts

```diff
@@ -0,0 +1,209 @@
+import { retrieveHolidays } from '$lib/api-holidays'
+import type { Holiday } from '$lib/api-holidays'
+
+type Kind = 'holiday'
+
+const capitalizeFirstLetter = (val) => {
+  return String(val).charAt(0).toUpperCase() + String(val).slice(1)
+}
+
+export const monthName = (date) => {
+  return capitalizeFirstLetter(date.toLocaleDateString('fr-FR', { month: 'long' }))
+}
+
+export class Item {
+  private _kind: Kind
+  private _title: String
+  private _description: String | null
+  private _date: Date | null
+  private _start_date: Date | null
+  private _end_date: Date | null
+
+  constructor(
+    kind: Kind,
+    title: String,
+    description: String | null,
+    date: Date | null = null,
+    start_date: Date | null = null,
+    end_date: Date | null = null
+  ) {
+    this._kind = kind
+    this._title = title
+    this._description = description
+    this._date = date
+    this._start_date = start_date
+    this._end_date = end_date
+  }
```

Comment:
I believe there's a shortcut to write this in typescript:

```suggestion
export class Item {
  constructor(
    private _kind: Kind,
    private _title: String,
    private _description: String | null,
    private _date: Date | null = null,
    private _start_date: Date | null = null,
    private _end_date: Date | null = null
  ) {}
```